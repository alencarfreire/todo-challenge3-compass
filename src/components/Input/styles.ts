import styled from "styled-components/native";
import { theme } from "../../theme/theme";

export const TextInput = styled.TextInput.attrs(() => ({
  placeholderTextColor: theme.colors.base.gray500,
}))`
  background-color: ${theme.colors.base.gray100};
  padding: 15px 18px;
  flex: 1;
  min-height: 52px;
  border: 2px solid;
  border-radius: 8px;
  font-family: ${theme.font.family.input};
  font-size: ${theme.font.size.input};
  line-height: ${theme.font.lineHeight.input};
  border-color: ${(props: { isFocused: boolean }) =>
    props.isFocused ? theme.colors.main.purpleBase : theme.colors.base.gray300};
`;
