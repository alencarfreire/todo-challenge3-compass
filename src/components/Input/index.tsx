import React, { useState } from "react";
import * as S from "./styles";

type Props = {
  placeholder?: string;
  value?: string;
  onChangeText?: (text: string) => void;
  secureTextEntry?: boolean;
  multiline?: boolean;
  error?: boolean;
};

export default function Input({
  placeholder,
  value,
  onChangeText,
  secureTextEntry,
  multiline,
  error,
}: Props) {
  const [isFocused, setIsFocused] = useState(false);
  return (
    <S.TextInput
      onChangeText={onChangeText}
      isFocused={isFocused}
      onFocus={() => setIsFocused(true)}
      onBlur={() => setIsFocused(false)}
      placeholder={placeholder}
      multiline={multiline}
      value={value}
      secureTextEntry={secureTextEntry}
      error={error}
    />
  );
}
