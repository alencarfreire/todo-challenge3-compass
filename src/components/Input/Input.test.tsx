import { fireEvent, render } from "@testing-library/react-native";
import Input from "./index";

describe("Component: Input", () => {
  it("should render the component", () => {
    render(<Input />);
  });

  it("should render the component with placeholder prop", () => {
    const placeholder = "Clique aqui";
    const { getByPlaceholderText } = render(
      <Input placeholder={placeholder} />
    );
    expect(getByPlaceholderText(placeholder)).toBeTruthy();
  });

  it("should update the value when typing", () => {
    const onChangeText = jest.fn();
    const initialValue = "Inicial";
    const newValue = "Novo valor";

    const { getByDisplayValue } = render(
      <Input value={initialValue} onChangeText={onChangeText} />
    );

    const input = getByDisplayValue(initialValue);
    fireEvent.changeText(input, newValue);
    expect(onChangeText).toHaveBeenCalledWith(newValue);
  });

  it("should update the isFocused state on focus and blur", () => {
    const { getByPlaceholderText } = render(
      <Input placeholder="Clique aqui" />
    );

    const input = getByPlaceholderText("Clique aqui");

    fireEvent(input, "focus");
    expect(input.props.isFocused).toBe(true);

    fireEvent(input, "blur");
    expect(input.props.isFocused).toBe(false);
  });

  it("should render with secureTextEntry prop", () => {
    const { getByPlaceholderText } = render(
      <Input placeholder="Senha" secureTextEntry />
    );

    const input = getByPlaceholderText("Senha");
    expect(input.props.secureTextEntry).toBe(true);
  });

  it("should render with multiline prop", () => {
    const { getByPlaceholderText } = render(
      <Input placeholder="Digite algo" multiline />
    );

    const input = getByPlaceholderText("Digite algo");
    expect(input.props.multiline).toBe(true);
  });

  it("should render without onChangeText prop", () => {
    const { getByPlaceholderText } = render(
      <Input placeholder="Sem handler" />
    );

    const input = getByPlaceholderText("Sem handler");
    expect(input.props.onChangeText).toBeUndefined();
  });

  it("should match the snapshot", () => {
    const { toJSON } = render(<Input placeholder="Clique aqui" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
