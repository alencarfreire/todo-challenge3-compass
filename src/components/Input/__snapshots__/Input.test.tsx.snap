// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Input should match the snapshot 1`] = `
<TextInput
  isFocused={false}
  onBlur={[Function]}
  onFocus={[Function]}
  placeholder="Clique aqui"
  placeholderTextColor="#6B6572"
  style={
    {
      "backgroundColor": "#F0EDF2",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderColor": "#E0DCE4",
      "borderStyle": "solid",
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "borderWidth": 2,
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "fontFamily": "Inter_400Regular",
      "fontSize": 16,
      "lineHeight": 22.4,
      "paddingBottom": 18,
      "paddingLeft": 18,
      "paddingRight": 18,
      "paddingTop": 18,
    }
  }
/>
`;
